packages/npm-dependencies-validation/src/index.ts

```diff
@@ -0,0 +1,68 @@
+import { spawnCommand } from "./utils/npmUtil";
+import { resolve, dirname } from "path";
+import { isManagedByNpm6 } from "./utils/packageJsonUtil";
+import {
+  DependencyIssue,
+  IssueType,
+  NpmLsDependency,
+  VscodeFsUri,
+} from "./types";
+
+const LS_DEPS = ["ls", "--depth=0"]; // dependencies and extraneous modules
+const LS_DEV_DEPS = [...LS_DEPS, "--dev"]; // devDependencies
+
+async function getDepsStatus(
+  packageJsonPath: string,
+  devDeps: boolean
+): Promise<any> {
+  // try to use DependencyType ???
+  return spawnCommand(
+    devDeps ? [...LS_DEV_DEPS] : [...LS_DEPS],
+    resolve(dirname(packageJsonPath))
+  );
+}
+
+function getProblemType(dependency: NpmLsDependency): IssueType | undefined {
+  if (dependency.missing) return "missing";
+  if (dependency.invalid) return "invalid";
+  if (dependency.extraneous) return "extraneous";
```

Comment:
- consider adding branch for **explicit** return of undefined
- what is better here (if or if / else) ?
- can you link to the blog post which recommand this style? it looks good, and I would want to read the discussion there

Reply:
Done
It was a video on Youtube in russian :)

Reply:
:)